File: backend/src/test/java/dev/coldhands/pair/stairs/backend/domain/DeveloperInfoTest.java
package dev.coldhands.pair.stairs.backend.domain;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class DeveloperInfoTest {

    @Test
    void comparable() {
        DeveloperInfo a = new DeveloperInfo(1, "a");
        DeveloperInfo b = new DeveloperInfo(1, "b");

        assertThat(a.compareTo(b)).isLessThan(0);
        assertThat(b.compareTo(a)).isGreaterThan(0);
        assertThat(a.compareTo(a)).isEqualTo(0);
    }
}

File: backend/src/test/kotlin/dev/coldhands/pair/stairs/backend/domain/developer/DeveloperInfoTest.kt
package dev.coldhands.pair.stairs.backend.domain.developer

import dev.coldhands.pair.stairs.backend.domain.DeveloperId
import io.kotest.matchers.comparables.shouldBeGreaterThan
import io.kotest.matchers.comparables.shouldBeLessThan
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test

class DeveloperInfoTest {

    @Test
    fun comparable() {
        val a = DeveloperInfo(
            id = DeveloperId(1),
            displayName = "a",
            archived = false
        )
        val b = a.copy(displayName = "b")

        a.compareTo(b) shouldBeLessThan 0
        b.compareTo(a) shouldBeGreaterThan 0
        a.compareTo(a) shouldBe 0
    }
}